src/hooks/achievement-rule-checker.js

```diff
@@ -24,9 +24,18 @@ module.exports = function (options = {}) {
         } else {
           achievementService.create({user_id: context.data.user_id, name: achievementRule.name, amount: 1});
         }
+
+        for (const replaceName of achievementRule.replaces) {
+          const replacedAchievement = await achievementService.find({
+            query: {
+              user_id: context.data.user_id,
+              name: replaceName
+            }
+          });
+          context.app.service('achievements').remove(replacedAchievement[0]._id);
```

Comment:
You should probably also `await` this, even if we do nothing with the result. This is to make sure it really is removed before continuing execution.